src/sonic-py-common/sonic_py_common/multi_asic.py

```diff
@@ -0,0 +1,290 @@
+    '''
+    The function connects to the config DB for a given namespace and
+    returns the handle
+    If no namespace is provided, it will connect to the db in the
+    default namespace.
+    In case of multi ASIC, the default namespace is the database
+    instance running the on the host
+
+    Returns:
+      handle to the config_db for a namespace
+    '''
+    SonicDBConfig.load_sonic_global_db_config()
+    config_db = ConfigDBConnector(namespace=namespace)
+    config_db.connect()
+    return config_db
+
+
+def connect_to_all_dbs_for_ns(namespace=DEFAULT_NAMESPACE):
+    '''
+    The function connects to the DBs for a given namespace and
+    returns the handle
+    If no namespace is provided, it will connect to the db in the
+    default namespace.
+    In case of multi ASIC, the default namespace is the
+    database instance running the on the host
+    In case of single ASIC, the namespace has to be DEFAULT_NAMESPACE
+
+    Returns:
+        handle to all the dbs for a namespaces
+    '''
+    SonicDBConfig.load_sonic_global_db_config()
+    db = SonicV2Connector(namespace=namespace)
+    for db_id in db.get_db_list():
+        db.connect(db_id)
+    return db
+
+
+def get_asic_conf_file_path():
+    """
+    Retrieves the path to the ASIC conguration file on the device
+
+    Returns:
+        A string containing the path to the ASIC conguration file on success,
+        None on failure
+    """
+    asic_conf_path_candidates = []
+
+    asic_conf_path_candidates.append(os.path.join(CONTAINER_PLATFORM_PATH,
+                                                  ASIC_CONF_FILENAME))
+
+    platform = get_platform()
+    if platform:
+        asic_conf_path_candidates.append(os.path.join(
+            HOST_DEVICE_PATH, platform, ASIC_CONF_FILENAME))
+
+    for asic_conf_file_path in asic_conf_path_candidates:
+        if os.path.isfile(asic_conf_file_path):
+            return asic_conf_file_path
+
+    return None
+
+
+def get_num_asics():
+    """
+    Retrieves the num of asics present in the multi ASIC platform
+
+    Returns:
+        Num of asics
+    """
+    asic_conf_file_path = get_asic_conf_file_path()
+
+    if asic_conf_file_path is None:
+        return 1
+
+    with open(asic_conf_file_path) as asic_conf_file:
+        for line in asic_conf_file:
+            tokens = line.split('=')
+            if len(tokens) < 2:
+                continue
+            if tokens[0].lower() == 'num_asic':
+                num_asics = tokens[1].strip()
+        return int(num_asics)
+
+
+def is_multi_asic():
+    """
+    Checks if the device is multi asic or not
+
+    Returns:
+        True: if the num of asic is more than 1
+    """
+    num_asics = get_num_asics()
+
+    return (num_asics > 1)
+
+
+def get_asic_id_from_name(asic_name):
+
+    if asic_name.startswith(ASIC_NAME_PREFIX):
+        return asic_name[len(ASIC_NAME_PREFIX):]
+    else:
+        return None
+
+
+def get_namespaces():
```

Comment:
i think we should rename API name here. It is little confusing with other API get_all_namespaces().
maybe get_linux_namespace_path() ?

Reply:
change the function name to get_namespaces_from_linux